metaseq_cli/train.py

```diff
@@ -486,6 +494,36 @@ def post_checkpoint_callback(cfg, filename):
             except (FileNotFoundError, AssertionError) as e:
                 logger.info(f"could not upload {filename}: {e}")
 
+        if do_evaluate:
+            _run_evaluations(
+                cfg.checkpoint.eval_command,
+                cfg.checkpoint.cloud_upload_path,
+                filename,
+                checkpoint_suffix,
+            )
+
+
+def _run_evaluations(eval_cmd, cloud_upload_path, local_file, checkpoint_suffix):
+    # WARNING: evals command needs to properly wait for
+    # all ranks to have finished uploading. Here we just
+    # wait for rank 0.
+    if distributed_utils.get_global_rank() != 0:
+        return
+    assert eval_cmd is not None, "--eval-command needs to be set."
+    cmd = eval_cmd.split()
+    checkpoint_name = local_file.split("/")[-1].replace(checkpoint_suffix, "")
```

Comment:
are we sure we're not going to be triggering this on both checkpoint_last and checkpoint_X?

Reply:
checkpoint_last is only written at the end of the training run. checkpoint_X is written at the update intervals. The logic here is to figure out which name we should use for evals. 

I didn't fully get your concern, does this address it?

https://github.com/facebookresearch/metaseq/blob/43815b8d4b08933f0c2af1cd275ec9b0108e89ba/metaseq/checkpoint_utils.py#L82-L100